### PR TITLE
Article: Python Function ID

### PR DESCRIPTION
**Pull request by Gouravchawla**
For issue #819

### CODE REVIEW

File: Python-Function-ID.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+## Argument
+
+### object
+
+The `object` argument can typically be a `int`,`float`, `str`,`list`, `dict`, `tuple` etc.
+
+## Return Value
+
+The return value would be a unique and constant `integer`.
+
```
Comment (theflametrooper) [inline Python-Function-ID.md]: Duplicate line can be removed.

File: Python-Function-ID.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Python id(object)
+
+`id()` is a built-in function in Python 3, which returns the *identity* of an object. The *identity* is a unique integer for that object during its lifetime. In CPython implementation, this is the address of the object in memory which is not necessarily true for other python implementations.
+
+## Argument
+
+### object
```
Comment (theflametrooper) [inline Python-Function-ID.md]: Change this to a `h4`

**Comment by Gouravchawla** [comment]
@theflametrooper I've implemented the changes. Could you check if that is what you suggested?

File: Python-Function-ID.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Python id(object)
+
+`id()` is a built-in function in Python 3, which returns the *identity* of an object. The *identity* is a unique integer for that object during its lifetime. In CPython implementation, this is the address of the object in memory which is not necessarily true for other python implementations.
+
+## Argument
+
+#### object
```
Comment (Rafase282) [inline Python-Function-ID.md]:
```diff
-#### object
+### object
```

File: Python-Function-ID.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+The `object` argument can typically be a `int`,`float`, `str`,`list`, `dict`, `tuple` etc.
+
+## Code Sample
```
Comment (Rafase282) [inline Python-Function-ID.md]:
```diff
-## Code Sample
+### Code Sample
```

**Comment by Rafase282** [comment]
I made some changes, there was no reason to go for h4 after a h2.

**Comment by Gouravchawla** [comment]
Thanks @Rafase282 . What else is required for this PR to be accepted?

**Comment by Rafase282** [comment]
From me, just that.

File: Python-Function-ID.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Python id(object)
+
+`id()` is a built-in function in Python 3, which returns the *identity* of an object. The *identity* is a unique integer for that object during its lifetime. In CPython implementation, this is the address of the object in memory which is not necessarily true for other python implementations.
```
Comment (alayek) [inline Python-Function-ID.md]: > In CPython implementation

A beginner would want to know what you mean by _CPython implementation_, so better link to the GitHub mirror of CPython interpreter, or some article that talks about this.

Comment (alayek) [inline Python-Function-ID.md]: > which is not necessarily true for other python implementations.

I know that CPython is not a perform-ant implementation. PyPy and Pyston would probably handle performance better by moving around objects in memory.

But reading it, it feels like that `id()` need to be same, is not part of the spec. Is that so? I am not fully aware of this.

**Comment by alayek** [comment]
Thanks for the article @Gouravchawla ! Great job :+1: 

I don't come across people often who care about underlying implementation details of a language. Just fix as per the comments on the article, and we are ready to merge.

**Comment by Gouravchawla** [comment]
Thanks @alayek . Also, I've made some changes and I think omitting the details about implementation makes more sense as the id() function is not used that much. Even the official documentation doesn't say much about it.

**Comment by Rafase282** [comment]
# LGTM :+1:

**Pull request merged**